apps/services/activities/src/app/worker/worker.module.ts

```diff
@@ -0,0 +1,48 @@
+import { DynamicModule, Module } from '@nestjs/common'
+import { SequelizeModule } from '@nestjs/sequelize'
+import { ConfigModule, ConfigType } from '@nestjs/config'
+import { BullModule as NestBullModule } from '@nestjs/bull'
+
+import { ActivitiesProcessor } from './activities.processor'
+import { createRedisCluster } from '@island.is/cache'
+import { SequelizeConfigService } from '../../sequelizeConfig.service'
+import { LoggingModule } from '@island.is/logging'
+import { Session } from '../sessions/session.model'
+import {
+  ActivitiesConfig,
+  activitiesQueueName,
+  bullModuleName,
+} from '../activities.config'
+import { SessionsService } from '../sessions/sessions.service'
+
+const BullModule = NestBullModule.registerQueueAsync({
+  name: activitiesQueueName,
+  useFactory: (config: ConfigType<typeof ActivitiesConfig>) => ({
+    prefix: `{${bullModuleName}}`,
+    createClient: () =>
+      createRedisCluster({
+        name: bullModuleName,
+        nodes: config.redis.nodes,
+        ssl: config.redis.ssl,
+        noPrefix: true,
+      }),
+  }),
+  inject: [ActivitiesConfig.KEY],
+})
```

Comment:
I would let the `SessionModule` do the bull module initialisation and import the `SessionModule` in the `WorkerModule`. You just list the `SessionService` and the `BullModule` in `exports` in the `SessionModule`.

Reply:
Aha, nice. Refactored.